### ISSUE TITLE
fetchQueue._promise can be referenced when fetchQueue has become undefined

### ISSUE THREAD

**Issue by jhudson8** [walmartlabs/thorax#304]
It seems that there might be a problem with https://github.com/walmartlabs/thorax/commit/3b945dfb1a8eb38e5ff0c071313ba417cb26fda2#diff-6c255b8f1d69c9f1b56a600c2a50c055

If an error occurs with the request, the error callback handler will eventually execute https://github.com/walmartlabs/thorax/blob/master/src/loading.js#L347 which will null out the fetchQueue.

This causes an issue with https://github.com/walmartlabs/thorax/blob/master/src/loading.js#L322 since the fetchQueue has just become undefined.

I assume this would only happen with a syncrhronous operation but, with a zepto ajaxJSONP call, its possible for this to error out: https://github.com/madrobby/zepto/blob/master/src/ajax.js#L117

While the code is correct (http jsonp request from https page), this masks the issue and makes it difficult to debug.
http://stackoverflow.com/questions/12216208/chrome-now-blocking-all-jsonp-requests-from-https-to-http
